core/src/main/scala/org/http4s/headers/Expires.scala

```diff
@@ -7,12 +7,29 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.parse.Parser
+import cats.syntax.all._
 import org.http4s.util.{Renderer, Writer}
 
 object Expires extends HeaderKey.Internal[Expires] with HeaderKey.Singleton {
   override def parse(s: String): ParseResult[Expires] =
-    HttpHeaderParser.EXPIRES(s)
+    parser.parseAll(s).leftMap { e =>
+      ParseFailure("Invalid Expires header", e.toString)
+    }
+
+  /* `Expires = HTTP-date` */
+  private[http4s] val parser: Parser[Expires] = {
+    import Parser.anyChar
+
+    def httpDate = HttpDate.parser
+
+    // A cache recipient MUST interpret invalid date formats, especially the
+    // value "0", as representing a time in the past (i.e., "already
+    // expired").
+    def invalid = anyChar.rep.as(HttpDate.Epoch)
```

Comment:
We used to handle  "0" and "-1" special, but the spec likes them all.

A better model in 1.0 would preserve the invalid value for round tripping instead of forcing it to the epoch.